sentry_sdk/integrations/httpx.py

```diff
@@ -52,13 +53,20 @@ def send(self, request, **kwargs):
             span.set_data("http.query", parsed_url.query)
             span.set_data("http.fragment", parsed_url.fragment)
 
-            for key, value in hub.iter_trace_propagation_headers():
-                logger.debug(
-                    "[Tracing] Adding `{key}` header {value} to outgoing request to {url}.".format(
-                        key=key, value=value, url=request.url
+            if should_propagate_trace(
+                str(request.url),
+                hub.client.options["trace_propagation_targets"]
+                if hub.client
```

Comment:
For what it's worth: Line 42 already "proves" there is a client. It is just a typing problem that you have to match on client here.

I was thinking that at some point, `hub.get_integration()` would be replaced with a method that returns both the integration and the current client so this check can be done all at once without typing issues

Reply:
Good Point. Have now changed the code to cast the hub.client. Now mypy is happy and the code is easier.

Reply:
And also good idea about having those checks in one place! 👍